src/Moryx.Products.Management/Implementation/RecipeManagement.cs

```diff
@@ -88,31 +89,37 @@ public IReadOnlyList<Workplan> LoadVersions(long workplanId)
         {
             using var uow = ModelFactory.Create();
 
-            var versions = new List<Workplan>();
             var repo = uow.GetRepository<IWorkplanRepository>();
-            do
+            var currentWorkplan = repo.GetAll()
```

Comment:
**Never** call `GetAll()` on a database. If you want to fetch data, filter and concat it, that do the majority of the work in Linq.